### PR TITLE
fix(grid): enable inline date filtering

### PR DESCRIPTION
**Pull request by jniles**
The inline date filters were all broken previously during an API change
that did not update the grids with the new API.  Additionally, the old
method did not work due to a syntax issue with the `moment().format()`
call.  This has been corrected by using an upper-cased version of the
bhConstant's date format.

Additionally, this commit bumps ui-grid's version to 4.0.4 to take
advantage of performance improvements.

**Example**
filterbydate
_Fig 1: Inline Filtering By Date_

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: client/src/partials/accounts/accounts.html [hunk 1]
```diff
@@ -40,12 +40,9 @@
     <div class="row">
       <div class="col-xs-12">
         <!-- @todo Consider how the application is loaded and cached and make this functionality apply to the whole application -->
-        <!-- Hide broken UI-Grid on page load -->
-        <div ng-if="!AccountsCtrl.pageLoaded"><i class="fa fa-spin fa-circle-o-notch"></i></div>
         <div
           id="account-grid"
-          style="height : calc(100vh - 65px)"
-          ng-show="AccountsCtrl.pageLoaded"
+          class="grid-full-height"
```
Comment (mbayopanda) [inline client/src/partials/accounts/accounts.html:45]: 👍

File: client/src/partials/accounts/accounts.js [hunk 2]
```diff
@@ -28,13 +28,14 @@ function AccountsController($rootScope, $timeout, AccountGrid, Notify, Constants
...
-  vm.loading = true;
 
   vm.Accounts = new AccountGrid();
   vm.Accounts.settup()
     .then(bindGridData)
     .catch(Notify.handleError)
-    .finally(toggleLoadingIndicator);
+    .finally(function () {
+      toggleLoadingIndicator();
+    });
```
Comment (mbayopanda) [inline client/src/partials/accounts/accounts.js]: What is the difference with `.finally(toggleLoadingIndicator);` syntax ? If there is no difference, `.finally(toggleLoadingIndicator);` seems more clear.

**Review by mbayopanda — APPROVED** [review]
LGTM

File: client/src/partials/accounts/accounts.js (hunk 2 above)
Comment (jniles) [reply client/src/partials/accounts/accounts.js]: They are logically equivalent.  I agree as well. :+1: I was trying to debug something and never switched it back.  Good catch :+1:

**Pull request merged**